### ISSUE TITLE
Review Intro Notebook 🤠 

### ISSUE THREAD

**Issue by jonathanferrari** [ds-modules/IB-C32#4]
# First Notebook Review 🥇📓📋 

## TL;DR
- review #3 and #1 then open the notebook via the link in the [README](https://github.com/ds-modules/IB-C32/tree/intro/intro)
- Leave detailed feedback on the pull request (#3) and a summary of your feedback on this thread

## Assignees:
@JessicaGolden @zcjanin @tanyagrawal

## Deadline:
<h3>Sunday, February 12<sup>th</sup>, 2023</h3>

## Action Items:
- ### Review:
    - The open issue for the creation of the introductory notebook: #1 
    - The pull request to merge the `intro` branch into `main`: #3 
    - Open the notebook in DataHub via the link in the [README](https://github.com/ds-modules/IB-C32/tree/intro/intro)
- ### Critique:
    - Please leave constructive criticism on the pull request as a comment
    - If there is something that you think should be fixed, how would you fix it?
    - The students who will be using the notebook have little to no coding experience
    - Overall impression of the notebook through the lens of the students who will use the notebook
- Leave a summary of your critique as a comment on this issue

## Details: 
Please be as detailed as possible! Any and all feedback is welcome, especially things that this notebook can do better or weaknesses that you see.

Examples of what feedback you can leave might be:
- typos that you found
- something was hard to understand (or would be for the students)
- was any of the formatting hard to follow
- was the notebook too long
- should I have gone more in-depth into anything
- is there anything I spent too much time explaining
- any other improvements you may have noticed

Of course, positive comments are more than welcome, but don't be afraid to give constructive feedback!

## Conclusion:
Thank you all, and if you have any questions/concerns/comments please leave them here, or in our slack channel; I will actively review both!

**Comment by tanyagrawal** [comment]
Overall Comment:

I really liked how comprehensive the notebook was and the detail in each section. If it was my first time coding, this notebook would make it easier to follow along. I also liked how there were examples for each concept that was introduced and how the code was broken down. It was also interactive and allowed students to experiment with concepts in the cells below. 

Feedback:

Under the section Notebook Structure, under cell types, I think we could add a code cell below the description to show the students what it will look like. 

Under Running Cells, we could specify that there are three different ways to run a cell in the first sentence

Under the Math section, on the last statement before the cells, there is a typo, "artound".

Under Assignment Statements, in the description under the first cell, I think "The final part is the left-hand side" should be "the right-hand side". I think it is referencing the "1+ 2+ 3+ 4" part of the assignment. I also have a suggestion for the description. 
"You can read the code from left to right starting at the x. In the code cell above, the x is the variable name and the equal sign assigns it to the expression on the right-hand side, 1 + 2 + 3 + 4. If we evaluate 1 + 2 + 3 + 4 it will equal 10. Therefore we can conclude that x = 1 + 2 + 3 + 4 is equivalent to x = 10. If you run the cell below, the output will be 10. "

Under the same section, in the description that states, "You can also chain...", we could remind the students that the value of x is stored in the notebook so when the expression is evaluated it will use x = 10. 

Under Lists, we can remind students that variables can store numbers, strings, etc., and that lists do not exclusively store numbers. 

Under the advanced/optional section in User-Defined Functions, there is a typo in the sentence before the blue colored note. "earier"

**Comment by JessicaGolden** [comment]
Comments:

Overall, I think this notebook would do an excellent job of introducing Jupyter and Python to someone with no experience. It has clear explanations and encourages experimentation, which is incredibly important when learning Python fundamentals.

- nit-picky but the red text says "advance" and I think it's supposed to be "advanced"
- my access was denied in the IPython link under "A Brief History"
- there's no mention of what to do if the hello world cell doesn't work (ie. the kernel died). Maybe it would be a good idea to mention what to do if the print statement doesn't work?
- once again a nitpick but the Errors section comment needs a space and capital W to be consistent (# We are missing...) not (#we are missing...) (also parenthesis is the singular form of parentheses)
- the seconds in a year example has a print statement regarding two hundred and thirty-four years but it is multiplied by 243 (not 234)
- the comment in the for loop section should have a space and capital D (# Do the intended...) not (#do the intended...)
- ^^^ same idea for loading the tips table, it should be # Here we see...
- also I'm pretty sure this isn't an issue, but some of the cells already show the output for me, and I want to make sure the outputs will be cleared before sending it to students
- I think there's an extra set of * around the quote in the "Notebooks in Practice" section

**Comment by zcjanin** [comment]
**Comments & Feedback:**

- For the learning outcomes, I think they are good but maybe they can be a little more descriptive and specific to provide more tangible objectives. 
- Access denied for iPython link
- This is nit picky but under "Saving and Loading" we can also mention that you can use command s
- Under the Math section, I think it would be helpful to include the examples after every operation that is introduced to help with flow and readability. 
- Under the errors section, it might be helpful to include two more examples of common errors to provide students with more exposure to other errors they may run into.
- In the Lists section, it would be helpful to introduce students how to make a list using brackets and that we can also have empty lists that can be populated. It would also be helpful to show the variation of lists that we can have (using strings, variables, etc.)
- In the Loops section, the word "repetitive" is spelled incorrectly.
- In the Modules section in the fourth paragraph, the word "could" should be replaced with "can" . In the fifth paragraph, after "However" it should read "this" and not "his". In the following sentence, the word "its" should be replaced with "it's".

Overall, I think this notebook is a great resource for introducing students with novice coding skills to Jupyter Notebooks and Python. I like the overall structure of the notebook and I think it flows really well. There are some minor grammatical errors, but I don't think there are any major issues with the notebook. I think that this was very comprehensive and provided good examples for each learning topic that students should learn.

**Comment by jonathanferrari** [comment]
All edits suggested by @tanyagrawal were accepted and added to the notebook in commit 39e44d2e89154596357a6462c9576deedb55c472 !

**Comment by jonathanferrari** [comment]
All edits suggested by @JessicaGolden were accepted and added to the notebook in commit 6ff865b94957f3a935a87e1de193b951633f2609!

**Comment by jonathanferrari** [comment]
All edits suggested by @zcjanin were accepted and added to the notebook in commit abe525c122ea50567fbe0315a1c2df436218cdba!
